docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
+- Run `yarn install` to install dependencies and run any required post-install scripts.
+- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass linting before you push a branch.
+
+## Testing
+
+### Running unit tests
+
+- Run `yarn workspace <workspaceName> run test` to run all tests for a package.
+- Run `yarn workspace <workspaceName> run jest --no-coverage <file>` to run a test file within the context of a package.
+- Run `yarn test` to run tests for all packages.
+
+> **Note**  
+> `workspaceName` here is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+### Integrating with other projects
+
+When developing changes to packages within this repository that a different project depends upon, you may wish to load those changes into the project and test them locally or in CI before publishing proper releases of those packages. To solve that problem, this repository provides a mechanism to publish "preview" versions of packages to GitHub Package Registry. These versions can then be used in the project like any other version, provided the project is configured to use that registry.
+
+#### As a MetaMask engineer
+
+If you're a MetaMask engineer, you can create these preview versions via draft pull requests:
+
+1. Navigate to your settings within GitHub and [create a classic access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @metamask:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Go to GitHub and open up a pull request for this repository (for the extension or mobile app, this needs to be a draft PR), then post a comment on the PR with the text `@metamask-bot preview-build`. (This triggers the `publish-preview` GitHub action.)
+4. After a few minutes, you will see a new comment indicating that all packages have been published with the format `<package name>-<commit id>`.
+5. Switch back to your project locally and update `package.json` by replacing the versions for the packages you've changed in your PR using the new version format (e.g. `1.2.3-e2df9b4` instead of `~1.2.3`), then run `yarn install`.
+6. Repeat steps 3-5 after pushing new changes to your PR to generate and use new preview versions.
+
+#### As an external contributor
```

Comment:
How about `As an independent developer`? Communicates the same message without the connotation of being "outside" or the necessity of "contributing" (even if that's what we hope is the result of reading this document of course ;))

Reply:
Fair! Updated in [8770a7a](https://github.com/MetaMask/controllers/pull/993/commits/8770a7a95c57c3936c8d5412c9a806a078fd9a2f).